Prvocisla

Cvachtacik opened the pull request.

Cvachtacik committed:
Delete example.py

Vymazanie vzorového súboru 'example.py'

Cvachtacik committed:
Create main.py

Vloženie hlavného programu

Cvachtacik committed:
Flake8

Úprava dĺžky komentárov v kóde, na odstránenie errorov z flake8.

Cvachtacik committed:
Doladenie zdrojového kódu

Odstránenie zvyšných errorov a warningov cez flake8

Cvachtacik committed:
Test 'prvocislo'

Test pre funkciu 'prvocislo' s testovacími hodnotami do 1000

Cvachtacik committed:
Test 'prvocislo_mr'

Test pre funkciu 'prvocislo_mr' s hodnotami väčšími ako 1000

turecekt commented on Prvočísla/main.py:
```diff
@@ -0,0 +1,120 @@
+    if prvocislo(cislo):
+        print(f"{cislo} je prvocislo. Pouzita deterministicka metoda.")
+    else:
+        print(f"{cislo} nie je prvocislo. Pouzita deterministicka metoda.")
```
toto celé dát pod podmínku if \_\_name\_\_ == '\_\_main\_\_':

Cvachtacik replied:
Čo sa týka zložky docs, mám tam html stránky (dokumentáciu vygenerovanú cez pdoc), ktroré sú na github-pages.

turecekt reviewed (changes requested):
- v Actions zrušte workflow pages-build-deployment (nastavit disable vpravo nahoře pod tlačítkem [...] zde: https://github.com/Cvachtacik/vs_project/actions/workflows/pages/pages-build-deployment)
- soubory main.py, test_prvocislo.py a test_pvocislo_mr.py přesunout do kořenového adresáře
- zrušit adresáře Prvočísla a PyDOC a docs včetně obsahu
- smazat soubory .DS_Store a main.html
- main.py upravit takto:

Cvachtacik committed:
PyDOC

Cvachtacik committed:
PyDOC

vygenerované html main.py

Cvachtacik committed:
Pdoc dokumentácia

Dokumentácia html vygenerovaná pomocou pdoc

Cvachtacik committed:
Create README.md

Cvachtacik committed:
Delete main.html

Cvachtacik committed:
Presunutie do korenového adresára

Cvachtacik committed:
Vymazanie zložky PyDOC

Cvachtacik committed:
Pridanie podmienky

Cvachtacik committed:
Delete .DS_Store

turecekt reviewed (changes requested):
- testy už fungují, ještě vyčistit kód, aby prošel přes statickou analýzu nástrojem flake8 viz chyby zde: https://github.com/Cvachtacik/vs_project/actions/runs/4029052153/jobs/6926551827
- v Actions zrušte spouštění workflow pages-build-deployment (nastavit disable vpravo nahoře pod tlačítkem [...] zde: https://github.com/Cvachtacik/vs_project/actions/workflows/pages/pages-build-deployment)
- odstranit adresáře .pytest_cache a docs včetně obsahu

Cvachtacik committed:
Úprava kódu - Flake8

Úprava kódu main.py, test_prvocislo.py a test_prvocislo_mr.py tak, aby vyhovoval testu Flake8

Cvachtacik committed:
Merge branch 'Prvocisla' of https://github.com/Cvachtacik/vs_project into Prvocisla

Cvachtacik committed:
Vymazanie adresárov

Vymazanie adresárov .pytest_cache a docs

turecekt commented on test_prvocislo.py:
```diff
@@ -0,0 +1,23 @@
+from main import prvocislo
+
+
+def test_prvocislo(self):
+    self.assertFalse(prvocislo(1))
```
```suggestion
def test_prvocislo():
    assert prvocislo(1) is False
```

Cvachtacik committed:
Zmena testov

turecekt reviewed (changes requested):
- v Actions opětovně povolit pouštění workflow Python checker(https://github.com/Cvachtacik/vs_project/actions/workflows/python_checker.yml) a naopak zrušit pouštění workflow pages-build-deployment (nastavit disable vpravo nahoře pod tlačítkem [...] zde: https://github.com/Cvachtacik/vs_project/actions/workflows/pages/pages-build-deployment)
- opravit kód tak, aby prošel přes workflow Python checker, tzn včetně flake8

Cvachtacik commented:
> * v Actions opětovně povolit pouštění workflow Python checker(https://github.com/Cvachtacik/vs_project/actions/workflows/python_checker.yml) a naopak zrušit pouštění workflow pages-build-deployment (nastavit disable vpravo nahoře pod tlačítkem [...] zde: https://github.com/Cvachtacik/vs_project/actions/workflows/pages/pages-build-deployment)
> * opravit kód tak, aby prošel přes workflow Python checker, tzn včetně flake8

Povolené púšťanie workflow Python checker, ale u pages-build-deployment nemám možnosť pre disable viz tu:
<img width="1403" alt="Screenshot 2023-01-29 at 00 23 48" src="https://user-images.githubusercontent.com/123431631/215295644-82985ae3-4533-45bc-8c50-9e41a9421793.png">

Cvachtacik committed:
Oprava

Oprava v teste

Cvachtacik committed:
Úprava komentárov

Úprava komentárov, aby vyhovovali pre flake8

Cvachtacik committed:
Oprava Update

Cvachtacik committed:
Update main.py

Cvachtacik committed:
Update main.py

Cvachtacik committed:
Update main.py

turecekt commented on main.py:
```diff
@@ -0,0 +1,119 @@
+import random
+
+
+def prvocislo(n):
```
```suggestion
def prvocislo(n: int) -> bool:
```

turecekt commented on main.py:
```diff
@@ -0,0 +1,119 @@
+    metóda zahŕňa delenie čísla n od 2 až do jeho druhej
+    odmocniny. Ak sa nedá deliť žiadnym číslom v tomto rozsahu,
+    potom sa považuje za prvočíslo.
+    """
```
```suggestion

    :type n: Testované číslo.
    :return: Existencia prvočísla.
    """
```

turecekt reviewed (changes requested):
Ještě poslední drobnosti, stejně doplnit u funkce prvocislo_mr.

Cvachtacik committed:
Update main.py

Posledné drobnosti

turecekt commented on main.py:
```diff
@@ -0,0 +1,122 @@
+    odmocniny. Ak sa nedá deliť žiadnym číslom v tomto rozsahu,
+    potom sa považuje za prvočíslo.
+
+    :type n: Testované číslo.
```
```suggestion
    :param n: Testované číslo.
```

turecekt commented on main.py:
```diff
@@ -0,0 +1,122 @@
+    return True
+
+
+def prvocislo_mr(n: int, k=5) -> bool:
```
```suggestion
def prvocislo_mr(n: int, k: int = 5) -> bool:
```

turecekt commented on main.py:
```diff
@@ -0,0 +1,122 @@
+    predchádzajúceho výpočtu, kým sa neobjaví výsledok
+    n-1 alebo kým sa nevykoná r-1 iterácií. Ak sa v cykle
+    nedosiahne výsledok n-1, číslo sa považuje za zložené.
+    """
```
```suggestion

    :param n: Testované číslo.
    :param k: Počet iterácií Miller-Rabin testu.
    :return: Existencia prvočísla.
    """
```

turecekt reviewed (changes requested):
poslední drobnost

Cvachtacik committed:
Update main.py

Posledné úpravy
